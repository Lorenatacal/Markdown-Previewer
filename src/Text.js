import React from 'react';
import ReactDOM from 'react-dom';
import Previewer from './Previewer';
import styled from 'styled-components';

const StyledContainer = styled.div`
    float:left;
    align: left;
    width: 45%;
    height: 400px;
    padding: 12px 20px;
    box-sizing: border-box;
    border: 2px solid #ccc;
    border-radius: 4px;
    background-color: #f8f8f8;
    resize: none;
    border-radius: 20px;
    padding: 0.25em;
    margin: 6em;
    margin-left: 25px;
    margin-right: 25px;
    align: center;
    text-align: center;
    font-family: arial;
`;

const StyledTitle = styled.h3`
    text-align: center;
`;

const StyledTextArea = styled.textarea`
    width: 80%;
    height: 70%;
`;

class Text extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: ''
        };

        this.handleChange = this.handleChange.bind(this);
    }


    handleChange(event) {
        this.setState({
            value: event.target.value
        });
    }

    render() {
        return (
            <StyledContainer>
                <StyledTitle>Text Area</StyledTitle>
                <StyledTextArea onChange={this.handleChange} />
            </StyledContainer>
        )
    }
}

export default Text;