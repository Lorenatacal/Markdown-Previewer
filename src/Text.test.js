import React, {Component} from 'react';
import Text from './Text';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

test("Text should render correctly", () => {
    const wrapper = Enzyme.shallow(<Text />)
    expect(toJson(wrapper)).toMatchSnapshot();
})