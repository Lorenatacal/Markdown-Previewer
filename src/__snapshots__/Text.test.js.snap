// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
<div>
  <styled.div>
    <styled.h3>
      Type your text in the box using the instructions below:
    </styled.h3>
    <styled.textarea
      onChange={[Function]}
      value="# Welcome to the Markdown Previewer
    
    ## The title was a header, this is a sub-header
    
    ### **Ussage:**
    
     * Write markdown text in this textarea.
     * See the changes in the previewer
    
    ### **Text attributes in number list:**
    
    1. *italic*,
    2. **bold**,
    3. **_italic-bold_**
    4. \`monospace\`
    5. ~~strikethrough~~ 
    
    *[Checkout my gitHub for more projects](https://github.com/Lorenatacal)*"
    />
  </styled.div>
  <Previewer
    input="# Welcome to the Markdown Previewer
  
  ## The title was a header, this is a sub-header
  
  ### **Ussage:**
  
   * Write markdown text in this textarea.
   * See the changes in the previewer
  
  ### **Text attributes in number list:**
  
  1. *italic*,
  2. **bold**,
  3. **_italic-bold_**
  4. \`monospace\`
  5. ~~strikethrough~~ 
  
  *[Checkout my gitHub for more projects](https://github.com/Lorenatacal)*"
  />
</div>
`;
